containers-contrib-readonly/src/Map/Contrib/Bench.hs

```diff
@@ -0,0 +1,39 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Map.Contrib.Bench where
+
+import Map
+import Gauge.Main (Benchmark, bench, defaultMain, whnf)
+import Relude hiding (fromList, Map)
+
+benchmark
+  :: forall k. (Enum k, Bounded k, Key k, NFData (Map k Int))
+  => String
+  -> Proxy k
+  -> IO ()
+benchmark label _ = do
+  let benchMaxElems = min 1024 (fromEnum (maxBound @k))
+  let upBound = foldl' (.) id (replicate benchMaxElems succ) minBound
```

Comment:
I think we can simplify benchmarks a lot if you pass lower and upper bounds to this function. So then in `main` we can just create variables:

```haskell
let lookupRangeLower = 0
let lookupRangeUpper = 1024
```

and pass them to this function. So the approach is more flexible because we can easily change the range and test with other range.

Reply:
Hmm great, this is the kinda-same thing we've done with the type alias `K`. Will do 👍